cassovary-core/src/main/scala/com/twitter/cassovary/util/SortedSeq.scala

```diff
@@ -0,0 +1,114 @@
+package com.twitter.cassovary.util
+
+import java.util
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * Sequence of sorted elements.
+ * 
+ * Assumes each element may appear in the Seq at most once.
+ */
+trait StrictlySortedSeqLike[T] extends mutable.IndexedSeq[T] {
```

Comment:
yeah, that's a lot of methods to override that are unnecessary for our purpose.

btw, do you still need this wrapper when moving to a graph-based design rather than a node-based design?